tools/c7n_gcp/c7n_gcp/filters/log_filter.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2019 Karol Lassak
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+
+from c7n.filters import FilterValidationError, ValueFilter
+from c7n.utils import chunks, local_session, type_schema, get_annotation_prefix
+from c7n_gcp.provider import resources as gcp_resources
+from datetime import datetime, timedelta
+from google.api_core.exceptions import TooManyRequests
+from google.cloud.logging import Client as LogClient
```

Comment:
we've actually tried to avoid pulling in google.cloud sdk because it brings a large dependency stack with it (protobuf, grpc, etc). the only exception where we use it was on c7n outputs where we would otherwise be duplicating their python logging implementation, but they've been something we've tried to avoid for resource/filter/action implementations.